code/modules/mod/modules/_module.dm

```diff
@@ -346,9 +348,16 @@
 		used_overlay = overlay_state_inactive
 	else
 		return
-	var/mutable_appearance/module_icon = mutable_appearance(overlay_icon_file, used_overlay, layer = standing.layer + 0.1)
+	var/mutable_appearance/module_icon
+	if(mask_worn_overlay)
+		var/icon/mod_mask = mod.generate_suit_mask()
+		mod_mask.Blend(icon(overlay_icon_file, used_overlay), ICON_MULTIPLY)
+		module_icon = mutable_appearance(mod_mask, layer = standing.layer + 0.1)
```

Comment:
are we using icons instead of filters for reasons of clientside performance? or something else. also how often is this blend called

Reply:
filters dont rotate